posthog/models/integration.py

```diff
@@ -191,6 +195,36 @@ def oauth_config_for_kind(cls, kind: str) -> OauthConfig:
                 id_path="sub",
                 name_path="email",
             )
+        elif kind == "snapchat":
+            if not settings.SNAPCHAT_APP_CLIENT_ID or not settings.SNAPCHAT_APP_CLIENT_SECRET:
+                raise NotImplementedError("Snapchat app not configured")
+
+            # snapchat only allows a single redirect URL
+            # we need to create a separate app for each region
+            if get_instance_region() == "EU":
+                return OauthConfig(
+                    authorize_url="https://accounts.snapchat.com/accounts/oauth2/auth",
+                    token_url="https://accounts.snapchat.com/accounts/oauth2/token",
+                    token_info_url="https://adsapi.snapchat.com/v1/me",
+                    token_info_config_fields=["me.id", "me.email"],
+                    client_id=settings.SNAPCHAT_APP_CLIENT_ID,
+                    client_secret=settings.SNAPCHAT_APP_CLIENT_SECRET,
+                    scope="snapchat-offline-conversions-api snapchat-marketing-api",
+                    id_path="me.id",
+                    name_path="me.email",
+                )
+            else:
+                return OauthConfig(
+                    authorize_url="https://accounts.snapchat.com/accounts/oauth2/auth",
+                    token_url="https://accounts.snapchat.com/accounts/oauth2/token",
+                    token_info_url="https://adsapi.snapchat.com/v1/me",
+                    token_info_config_fields=["me.id", "me.email"],
+                    client_id=settings.SNAPCHAT_APP_CLIENT_ID,
+                    client_secret=settings.SNAPCHAT_APP_CLIENT_SECRET,
+                    scope="snapchat-offline-conversions-api snapchat-marketing-api",
+                    id_path="me.id",
+                    name_path="me.email",
+                )
```

Comment:
How do these differ?

Reply:
Right - missed this when removing the `SNAPCHAT_EU_APP` variable. Removed the condition.